### PR TITLE
Updates reachabiltiy_validator to check for all tool_frames

### PR DESCRIPTION
**Pull request by Leusmann**
Updates the reachability_validator to check all tool_frames for reach ability, keeps the retrace pose in mind

### CODE REVIEW

**Comment by Leusmann** [comment]
Nice find. I can reproduce the error on my side. Will fix

**Comment by Leusmann** [comment]
I think I found a solution for this problem, which is a bit surprising. The IK Server only finds solutions if the position was given in the world frame? 
Could somebody please check if it works for them as well?

File: src/pycram/pose_generator_and_validator.py [hunk 1]
```diff
@@ -153,36 +154,59 @@ def reachability_validator(pose: Pose,
...
         except IKError:
             pass
         finally:
             robot.set_joint_states(joint_state_before_ik)
+    if arms:
+        res = True
     return res, arms
+
+
+def collision_check(robot, allowed_collision):
```
Comment (Tigul) [inline src/pycram/pose_generator_and_validator.py:192]: Doc please.

**Comment by Leusmann** [comment]
Added some documentation strings - should be better now

**Pull request merged**